app/ui/legacy/src/main/java/com/fsck/k9/ui/settings/general/GeneralSettingsFragment.kt

```diff
@@ -33,8 +69,32 @@ class GeneralSettingsFragment : PreferenceFragmentCompat() {
         }
     }
 
+    private fun observeEvents() {
+        viewModel.events.observeEvents(this) { event ->
+            when (event) {
+                is GeneralSettingsEvent.ExportLogsMenuEnabled -> setExportLogsMenuEnabled(event.isEnabled)
+                GeneralSettingsEvent.ExportLogsSuccess -> displayExportLogsSuccess()
+                GeneralSettingsEvent.ExportLogsFailure -> displayExportLogsFailure()
```

Comment:
None of these should be one-time events. We want the menu enabled state to be set again when the Activity/Fragment is recreated. Also, we want the Snackbars to be recreated if necessary.

Back to the drawing board :sob: